src/wp-includes/functions.php

```diff
@@ -824,7 +824,7 @@ function wp_extract_urls( $content ) {
 			. '(?:'
 				. '\([\w\d]+\)|'
 				. '(?:'
-					. "[^`!()\[\]{};:'\".,<>«»“”‘’\s]|"
+					. "[^`!()\[\]{}:'\".,<>«»“”‘’\s]|"
```

Comment:
I'm concerned about the backwards-compatibility break being introduced here. Any URLs that contain a `;` will break as the `;` will be retained.

For example for `https://example.org;`:
- Current behavior: `https://example.org` is returned
- With this change: `https://example.org;` is returned
 
This change changes these URLs. While the change fixes the original reported issue, it causes another.

Instead, I'd suggest identifying and excluding only the `&amp;` pattern. Makes the regex more complicated. But retains back-compat and fixes the reported issue.

Reply:
Thank you for the feedback, @hellofromtonya!

At this time, `wp_extract_urls` is referenced in 2 places:

- `pingback` at `/wp-includes/comment.php:3019`
- `do_enclose` at `/wp-includes/functions.php:879`

The `;` is a [reserved character](https://www.rfc-editor.org/rfc/rfc3986#section-2.2) that [can be used in a path](https://www.rfc-editor.org/rfc/rfc3986#section-3.3). For example: `http://example.org/file;v=1.1`.

WordPress doesn't let users save posts with slugs like that, and would reformat attempts like `file;v=1.1` => `filev1-1`, so this isn't likely to affect `pingback`.

However, `wp_extract_urls` is also used for [feed media file enclosures](https://developer.wordpress.org/reference/functions/rss_enclosure/) in `do_enclose`, where it's possible a remote audio/video file could utilize `;` in the path 🤔. This isn't likely for WordPress-hosted media, but an external hosting service might.

> Instead, I'd suggest identifying and excluding only the `&amp;` pattern.

In addition to `&`, there are several other reserved characters that could be legitimately encoded in the URL. In effect this would be a filter on allowable query string entities. For just WordPress (`pingback`), we're only expecting `&`, but external media files could be different.

There are also three flavors for HTML encoding to consider. Using `&` as an example: **named** `&amp;`, **ascii** `&#38;`, and **hex** `&#x26;` -- and the ascii/hex options can have leading zeros. It would definitely get more complicated!

---

Here's a try at what the ultimate impacts might be -- please add/comment if you see anything else:

- `pingback`: Existing posts: WordPress would have already tracked the inadvertently "fixed" version of the URL, so old links that were successfully "pung" would not be affected. For new/updated posts, the now [originally] broken URL would be excluded from future pingback attempts. **Impact:** For new posts, prevents pingback attempts for broken URLs.
- `do_enclose`: For existing posts, the old "fixed" URL would remain as an enclosure until the post was updated (this is stored in post meta). For new or updated posts, the "fixed" URL would be removed from the list of video/audio enclosures. **Impact:** For new/updated posts, removal of broken audio/video links from enclosure meta (and resulting feed data). Potential addition of excluded audio/video links (e.g. with `;` in path) to enclosure meta.

Thoughts? How deeply should we cover these corner cases?

Reply:
@hellofromtonya, I received a suggestion from @costdev regarding keeping backward compatibility for the semicolon removal, and have updated `wp_extract_urls` accordingly. Also updated tests with specific test for the compatibility concern.